lib/tensorflow/tensor.rb

```diff
@@ -182,17 +182,39 @@ def ruby_array_to_c(array, type)
   #   - Value of the element contained in the specified position in the tensor.
   #
   def getval(dimension)
-    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != dimensions.length
+    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != shape.length
     (0..dimension.length-1).each do |i|
-      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > dimensions[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
+      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > shape[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
     end
-    sum = dimension[dimension.length - 1]  - 1
-    prod = dimensions[dimensions.length - 1]
+    sum = dimension.last - 1
+    prod = shape.last
     (0..dimension.length - 2).each do |i|
        sum += (dimension[dimension.length - 2 - i] - 1) * prod
-       prod *= dimensions[dimensions.length - 2 - i]
+       prod *= shape[shape.length - 2 - i]
     end
 
     flatten[sum]
   end
+
+  private
+
+  #
+  # Recursively finds the shape of the input array.
+  #
+  # * *Returns* :
+  #   - Dimension array `[[2], [4]].shape` => `[2, 1]`
+  #
+  # TODO: Handle non-rectangular arrays and raise error if mixed data types
+  def self.set_shape(value)
```

Comment:
Why call it **set_**? It doesn't set's anything (not speaking of the fact that even if it had, in Ruby we never use `set_`/`get_` prefixes), just analyzes value. For me, it is something like `guess_shape` or `shape_of(value)`